src/System.Memory/src/System/SpanHelpers.byte.cs

```diff
@@ -178,5 +178,76 @@ public static unsafe bool SequenceEqual(ref byte first, ref byte second, int len
         NotEqual: // Workaround for https://github.com/dotnet/coreclr/issues/9692
             return false;
         }
+
+        public static unsafe bool StartsWith(ref byte searchSpace, int searchSpaceLength, ref byte value, int valueLength)
+        {
```

Comment:
Sounds good. I measured the perf difference and it is quite large if I don't use SequenceEqual.